src/task.rs

```diff
@@ -7,6 +7,46 @@ use esp_idf_sys::*;
 
 use crate::delay::TickType;
 use crate::interrupt;
+use crate::watchdog::task::TaskWatchdog;
+
+#[derive(Debug)]
+pub struct Task(TaskHandle_t);
```

Comment:
For me, the biggest question is whether this API is safe. As in what happens if the task is over (as in the `TaskHandle_t` structure is invalid as the underlying task was dropped/deleted) - yet - it is still attached to the watchdog? Would it crash?